src/main/scala/sigmastate/eval/RuntimeCosting.scala

```diff
@@ -1489,26 +1493,26 @@ trait RuntimeCosting extends CostingRules with DataCosting with Slicing { IR: Ev
       case CalcBlake2b256(In(input)) =>
         val bytesC = asRep[Costed[Coll[Byte]]](input)
         val res = sigmaDslBuilder.blake2b256(bytesC.value)
-        val cost = opCost(Seq(bytesC.cost), perKbCostOf(node, bytesC.size.dataSize))
+        val cost = opCost(res, Seq(bytesC.cost), perKbCostOf(node, bytesC.size.dataSize))
         mkCostedColl(res, Blake2b256.DigestSize, cost)
       case CalcSha256(In(input)) =>
         val bytesC = asRep[Costed[Coll[Byte]]](input)
         val res = sigmaDslBuilder.sha256(bytesC.value)
-        val cost = opCost(Seq(bytesC.cost), perKbCostOf(node, bytesC.size.dataSize))
+        val cost = opCost(res, Seq(bytesC.cost), perKbCostOf(node, bytesC.size.dataSize))
         mkCostedColl(res, Sha256.DigestSize, cost)
 
       case utxo.SizeOf(In(xs)) =>
         xs.elem.eVal.asInstanceOf[Any] match {
           case ce: CollElem[a,_] =>
             val xsC = asRep[Costed[Coll[a]]](xs)
             val v = xsC.value.length
-            RCCostedPrim(v, opCost(Seq(xsC.cost), costOf(node)), SizeInt)
+            RCCostedPrim(v, opCost(v, Seq(xsC.cost), costOf(node)), SizeInt)
           case se: StructElem[_] =>
             val xsC = asRep[Costed[Struct]](xs)
-            RCCostedPrim(se.fields.length, opCost(Seq(xsC.cost), costOf(node)), SizeInt)
+            RCCostedPrim(se.fields.length, opCost(xsC.value, Seq(xsC.cost), costOf(node)), SizeInt)
```

Comment:
`se.fields.length` should go to opCost, and remember this is graph building, so compiler cannot do common subexpression optimization, so better introduce local val and use it twice.

Reply:
Fixed.